--- a/src/wire/ipv6.rs	2026-06-24 23:16:09.014329167 +0800
+++ b/src/wire/ipv6.rs	2026-06-17 10:32:35.708515373 +0800
@@ -122,6 +122,7 @@
     /// Query whether the IPv6 address is a [solicited-node multicast address].
     ///
     /// [Solicited-node multicast address]: https://datatracker.ietf.org/doc/html/rfc4291#section-2.7.1
+    #[allow(dead_code)]
     fn is_solicited_node_multicast(&self) -> bool;
 
     /// If `self` is a CIDR-compatible subnet mask, return `Some(prefix_len)`,
@@ -199,6 +200,7 @@
         }
     }
 
+    #[allow(dead_code)]
     fn is_solicited_node_multicast(&self) -> bool {
         self.octets()[0..13]
             == [
